File: src/main/java/net/bms/genera/capability/interfaces/IFaerieInformation.java
package net.bms.genera.capability.interfaces;

import net.minecraft.nbt.NBTTagCompound;

/**
 * Created by benjaminsutter on 7/3/17.
 */
public interface IFaerieInformation {
    void setType(String type);
    String getType();

    void setSize(float size);
    float getSize();

    void setMaxHealth(double maxHealth);
    double getMaxHealth();

    void setLevel(int level);
    int getLevel();

    void setCurrentExp(int exp);
    int getCurrentExp();

    void deserializeNBT(NBTTagCompound nbt);
    NBTTagCompound serializeNBT();
}

File: src/main/kotlin/net/bms/genera/capability/interfaces/IFaerieInformation.kt
package net.bms.genera.capability.interfaces

import net.minecraft.nbt.NBTTagCompound

interface IFaerieInformation {
    var type: String
    var size: Float
    var maxHealth: Double
    var level: Int
    var currentExp: Int

    fun deserializeNBT(nbt: NBTTagCompound)
    fun serializeNBT(): NBTTagCompound
}
